File: library/src/main/java/org/coner/crispy_fish/filetype/ecf/EventControlFile.java
package org.coner.crispy_fish.filetype.ecf;

import java.nio.file.Path;

public class EventControlFile {

    private final EventControlFileAssistant ecfAssistant;
    private final Path path;
    private final boolean twoDayEvent;

    public EventControlFile(Path path, boolean twoDayEvent) {
        ecfAssistant = new EventControlFileAssistant();
        if (!ecfAssistant.isEventControlFilePath(path)) {
            throw new NotEventControlFilePathException(path);
        }
        this.path = path;
        this.twoDayEvent = twoDayEvent;
    }

    public Path getPath() {
        return path;
    }

    public boolean isTwoDayEvent() {
        return twoDayEvent;
    }
}

File: library/src/main/kotlin/org/coner/crispy_fish/filetype/ecf/EventControlFile.kt
package org.coner.crispy_fish.filetype.ecf

import org.crispy_fish.filetype.ecf.NotEventControlFilePathException
import java.nio.file.Path

class EventControlFile(val path: Path, val isTwoDayEvent: Boolean, val conePenalty: Int) {

    private val ecfAssistant: EventControlFileAssistant

    init {
        ecfAssistant = EventControlFileAssistant()
        if (!ecfAssistant.isEventControlFilePath(path)) {
            throw NotEventControlFilePathException(path)
        }
    }
}
